### PR TITLE
Deduplicate README.md

### LINKED ISSUE #71: Deduplicate `README.md` files in root and datafusion directory
**Describe the bug**
The `README.md` files in the root and in the `datafusion` directory are (mostly) copies of each other.
There are small differences between the two, e.g. added functionality that is not included.

**Expected outcome **
Differences between the two are removed. Unneeded details in the root readme should be avoided.

### PR DESCRIPTION
**Pull request by msathis**
# Which issue does this PR close?

Closes #71 .

 # Rationale for this change

Removing duplicated content

# What changes are included in this PR?

Remove duplicated content & move Developers.md to root

# Are there any user-facing changes?
N/A

### CODE REVIEW

**Comment by Dandandan** [comment]
Thanks @msathis ! just an idea, maybe we can point the datafusion crate to the readme in the root too?

https://doc.rust-lang.org/cargo/reference/manifest.html#the-readme-field

**Comment by msathis** [comment]
> Thanks @msathis ! just an idea, maybe we can point the datafusion crate to the readme in the root too?

Nice idea 💡  @Dandandan. Never knew this. Will do it 👌

**Review by Dandandan — APPROVED** [review]
Awesome, looks good! 😎

File: README.md [hunk 1]
```diff
@@ -141,11 +141,11 @@ DataFusion also includes a simple command-line interactive SQL utility. See the
...
- - [x] Constant folding
- - [x] Join Reordering
- - [x] Limit Pushdown
- - [x] Projection push down
- - [x] Predicate push down
+- [x] Constant folding
+- [x] Join Reordering
+- [x] Limit Pushdown
+- [x] Projection push down
+- [x] Predicate push down
```
Comment (alamb) [inline README.md:148]: I think these were meant to be sub bullets (they are features of the query optimizer) but we can fix that in a subsequent PR)

**Review by alamb — APPROVED** [review]
Thanks @msathis !

**Pull request merged**